app/src/parser.c

```diff
@@ -93,27 +106,241 @@ parser_error_t parser_getItem(const parser_context_t *ctx, uint8_t displayIdx, c
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        case 5:
+            snprintf(outKey, outKeyLen, "Nonce");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->nonce) != NULL) {
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        default:
+            return parser_no_data;
+    }
+
+    return parser_ok;
+}
+
+parser_error_t printWalletSpawn(const parser_context_t *ctx, uint8_t displayIdx, char *outKey, uint16_t outKeyLen,
+                                char *outVal, uint16_t outValLen, uint8_t pageIdx, uint8_t *pageCount) {
+    char buff[50] = {0};
+    switch (displayIdx) {
+        case 0:
+            snprintf(outKey, outKeyLen, "Tx type");
+            snprintf(outVal, outKeyLen, "Spawn");
+            break;
+        case 1:
+            snprintf(outKey, outKeyLen, "Principal");
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->principal.ptr), ADDRESS_LENGTH, pageIdx, pageCount);
+            break;
+        case 2:
+            snprintf(outKey, outKeyLen, "Pubkey");
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->spawn.wallet.pubkey.ptr), PUB_KEY_LENGTH, pageIdx,
+                          pageCount);
+            break;
+        case 3:
+            snprintf(outKey, outKeyLen, "Account template");
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->spawn.account_template.ptr), ADDRESS_LENGTH,
+                          pageIdx, pageCount);
+            break;
+
+        case 4:
+            snprintf(outKey, outKeyLen, "Gas price");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->gas_price) != NULL) {
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        case 5:
+            snprintf(outKey, outKeyLen, "Nonce");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->nonce) != NULL) {
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        default:
+            return parser_no_data;
+    }
+
+    return parser_ok;
+}
 
+parser_error_t printMultisigSpawn(const parser_context_t *ctx, uint8_t displayIdx, char *outKey, uint16_t outKeyLen,
+                                  char *outVal, uint16_t outValLen, uint8_t pageIdx, uint8_t *pageCount) {
+    char buff[50] = {0};
     switch (displayIdx) {
         case 0:
-            snprintf(outKey, outKeyLen, "Blind");
-            snprintf(outVal, outValLen, "Signing");
+            snprintf(outKey, outKeyLen, "Tx type");
+            snprintf(outVal, outKeyLen, "Spawn");
```

Comment:
we may need to ask if they prefer to see "Multisig Spawn" or similar

Reply:
yes, this can help to identify the type of account. I will add this